### PR TITLE
Luxor/reward-curve-proposal

### PR DESCRIPTION
**Pull request by dobertRowneySr**
addresses: https://github.com/Joystream/joystream/issues/3702

### CODE REVIEW

File: runtime-modules/council/src/lib.rs [hunk 1]
```diff
@@ -367,6 +368,10 @@ decl_storage! { generate_storage_info
 
         /// Councilor reward per block
         pub CouncilorReward get(fn councilor_reward) config(): Balance<T>;
+
+        /// Era payou damping factor: a parameter in [0,1] that can be used to reduce the era
+        /// payout without changing the reward curve directly
+        pub EraPayoutDampingFactor get(fn era_payout_damping_factor) config(): Percent;
```
Comment (mnaamani) [inline runtime-modules/council/src/lib.rs]: We probably want the default/initial damping factor to be 100%

File: runtime/src/lib.rs [hunk 2]
```diff
@@ -536,11 +537,21 @@ impl EraPayout<Balance> for NoInflationIfNoEras {
             // PoA mode: no inflation.
             (0, 0)
         } else {
-            <pallet_staking::ConvertCurve<RewardCurve> as EraPayout<Balance>>::era_payout(
-                total_staked,
-                total_issuance,
-                era_duration_millis,
-            )
+            // rescale the era payout according to the damping factor while keeping the invariant
+            // era_payou + rest = max_payout
```
Comment (mnaamani) [inline runtime/src/lib.rs:541]: By keeping this invariant, the damping factor is only reducing the payout to the validator, the rest, which increases if damping factor is less than 100% goes to the nominators. So we are not reducing the inflation only payout amount to validators. Is this the intention?
If we wish to reduce the overall inflation, then the max_payout needs to be scaled as well.

Comment (mnaamani) [reply runtime/src/lib.rs:541]: Sorry my mistake, the staking pallet will send the "remainder" (rest) amount to be handled by `RewardRemainder` trait (not for nominators). We have set this trait as empty tuple `()`, so the amount will be discarded.

File: runtime-modules/council/src/lib.rs (hunk 1 above)
Comment (dobertRowneySr) [reply runtime-modules/council/src/lib.rs]: I believe I have set it to 100% to the chainspec file

File: runtime-modules/council/src/lib.rs [hunk 3]
```diff
@@ -843,6 +859,64 @@ decl_module! {
...
+        ///
+        /// # <weight>
+        ///
+        /// ## weight
+        /// `O (1)`
+        /// - db:
+        ///    - `O(1)` doesn't depend on the state or parameters
+        /// # </weight>
+        #[weight = 10_000_000] // TODO: adjust
+        pub fn decrease_council_budget(origin, reduction_amount: Balance<T>) -> Result<(), Error<T>> {
```
Comment (freakstatic) [inline runtime-modules/council/src/lib.rs]: @ignazio-bovo are you sure you want to include this change?
seems to be out of scope

**Review by dobertRowneySr — COMMENTED** [review]
I think I still have to add benchmarks for this but at this point I would wait before 
- Council reduce budget
- Budget WG spending

File: runtime-modules/council/src/lib.rs (hunk 1 above)
Comment (mnaamani) [reply runtime-modules/council/src/lib.rs]: Yes I see it. We will need to add a runtime migration to also set it to 100%.

File: runtime-modules/council/src/lib.rs (hunk 3 above)
Comment (mnaamani) [reply runtime-modules/council/src/lib.rs]: I think its because he is building on prior PRs

File: types/src/augment/augment-api-consts.ts [hunk 4]
```diff
@@ -585,10 +585,6 @@ declare module '@polkadot/api-base/types/consts' {
        * Exports `Set Councilor Reward Proposal Parameters` proposal parameters.
        **/
       setCouncilorRewardProposalParameters: PalletProposalsEngineProposalParameters & AugmentedConst<ApiType>;
-      /**
-       * Era payout damping factor
-       **/
-      setEraPayoutDampingFactorProposalParameters: PalletProposalsEngineProposalParameters & AugmentedConst<ApiType>;
```
Comment (thesan) [inline types/src/augment/augment-api-consts.ts]: Hey @ignazio-bovo this last commit removes the set era payout damping factor proposal types to add the update token pallet governance ones, is it a mistake ?

FYI It also makes the network tests fail (including the chain setup scenario) so I had to rebase the QN work on the previous commit.

File: runtime-modules/council/src/lib.rs (hunk 1 above)
Comment (mnaamani) [reply runtime-modules/council/src/lib.rs]: Prepared this https://github.com/ignazio-bovo/joystream/pull/14

File: types/src/augment/augment-api-consts.ts (hunk 4 above)
Comment (mnaamani) [reply types/src/augment/augment-api-consts.ts]: Yeah the PR needs to be rebased on luxor and types regenerated.

File: runtime-modules/council/src/lib.rs (hunk 3 above)
Comment (dobertRowneySr) [reply runtime-modules/council/src/lib.rs]: Constraints:
- Zero is allowed
- Max percent, currently 100%, potentially to increasing to above 100%

Comment (dobertRowneySr) [reply runtime-modules/council/src/lib.rs]: We want this to be above 100% :(

**Comment by kdembler** [comment]
Let's allow the multiplier to be 0.1-2.0. Mokhtar suggests to rename from damping to just multiplier

**Comment by ignazio-bovo** [comment]
add the init default value

File: runtime-modules/council/src/lib.rs (hunk 1 above)
Comment (mnaamani) [inline runtime-modules/council/src/lib.rs]: @ignazio-bovo suggested this might be simpler than doing a migration

```suggestion
        pub EraPayoutDampingFactor get(fn era_payout_damping_factor) config(): Percent = Percent::from_percent(100);
```

Comment (mnaamani) [reply runtime-modules/council/src/lib.rs:374]: @ignazio-bovo suggested this might be simpler than doing a migration

File: tests/network-tests/src/misc/postRuntimeUpgrade.ts [hunk 5]
```diff
@@ -64,5 +64,9 @@ export default async function assertValues({ api }: FlowProps): Promise<void> {
   // assert.deepEqual(curatorGroup.permissionsByLevel, expectedPermissions)
   assert.equal(`${curatorGroup.permissionsByLevel}`, `${expectedPermissions}`)
 
+  // Damping factor after runtime should be 100% = default
```
Comment (mnaamani) [inline tests/network-tests/src/misc/postRuntimeUpgrade.ts:15]: Some of the previous assertions in this function need to be removed or fixed, they were for ephesus -> nara upgrade. As well as the setup in the `preRuntimeUpgrade.ts`

**Review by mnaamani — CHANGES_REQUESTED** [review]

File: runtime/src/proposals_configuration/staging.rs [hunk 6]
```diff
@@ -372,6 +372,21 @@ pub(crate) fn update_token_pallet_token_governance_parameters(
     }
 }
 
+// TODO (luxor/reward-curve-proposal): adjust parameters
+// Proposal parameters for the 'Set Era Payout Damping Factor' proposal
+pub(crate) fn set_era_payout_damping_factor() -> ProposalParameters<BlockNumber, Balance> {
+    ProposalParameters {
+        voting_period: days!(3),
```
Comment (mnaamani) [inline runtime/src/proposals_configuration/staging.rs]: `days` macro is not imported. But I think for staging profile we probably want minutes here.

**Review by mnaamani — APPROVED** [review]
Ok LGTM.
Changes to runtime-upgrade integration test do not look complete and they will be addressed separately as per https://github.com/Joystream/joystream/issues/5109
